homeassistant/components/config/device_registry.py

```diff
@@ -0,0 +1,47 @@
+"""HTTP views to interact with the device registry."""
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import async_get_registry
+from homeassistant.components import websocket_api
+
+DEPENDENCIES = ['websocket_api']
+
+WS_TYPE_LIST = 'config/device_registry/list'
+SCHEMA_WS_LIST = websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
+    vol.Required('type'): WS_TYPE_LIST,
+})
+
+
+async def async_setup(hass):
+    """Enable the Entity Registry views."""
+    hass.components.websocket_api.async_register_command(
+        WS_TYPE_LIST, websocket_list_devices,
+        SCHEMA_WS_LIST
+    )
+    return True
+
+
+@callback
+def websocket_list_devices(hass, connection, msg):
+    """Handle list devices command.
+
+    Async friendly.
+    """
+    async def retrieve_entities():
+        """Get devices from registry."""
+        registry = await async_get_registry(hass)
+        connection.send_message_outside(websocket_api.result_message(
+            msg['id'], [{
+                'config_entries': list(entry.config_entries),
+                'connections': list(entry.connections),
+                'identifiers': list(entry.identifiers),
```

Comment:
Are connections and identifiers really relevant to expose to the Gui? Config entries I can understand to tie be able to tie them together in some hierarchy but the others are mostly internal usage.

Reply:
Good point, little too eager!

Reply:
Actually connections could be beneficial? you can see mac address etc ?

Reply:
If we want to expose connections and identifiers we need to up our game and standardise what can be a part of them. Right now it is not concrete enough.

Reply:
Well if we do expose connections we can just as well keep identifiers as well.

Reply:
haha.

Look, connections is actually standardized to an extend:

https://github.com/home-assistant/home-assistant/blob/d076251b189c0a11f2bd528fdfdd0fb3d5dc8f5d/homeassistant/helpers/device_registry.py#L20-L21